internal/servers/controller/internal/metric/cluster.go

```diff
@@ -0,0 +1,177 @@
+package metric
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/boundary/globals"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/observability/event"
+	"github.com/prometheus/client_golang/prometheus"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
+)
+
+const (
+	invalidValue = "invalid"
+
+	labelGRpcCode    = "grpc_code"
+	labelGRpcService = "grpc_service"
+	labelGRpcMethod  = "grpc_method"
+	clusterSubSystem = "controller_cluster"
+)
+
+var (
+	gRpcMsgSizeBuckets = prometheus.ExponentialBuckets(100, 10, 8)
+
+	// gRpcRequestLatency collects measurements of how long it takes
+	// the boundary system to reply to a request to the controller api
+	// from the time that boundary received the request.
+	gRpcRequestLatency prometheus.ObserverVec = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: globals.MetricNamespace,
+			Subsystem: clusterSubSystem,
+			Name:      "grpc_request_duration_seconds",
+			Help:      "Histogram of latencies for HTTP requests.",
+			Buckets:   prometheus.DefBuckets,
+		},
+		[]string{labelGRpcCode, labelGRpcService, labelGRpcMethod},
+	)
+
+	// gRpcRequestSize collections measurements of how large each request
+	// to the boundary controller cluster is.
+	gRpcRequestSize prometheus.ObserverVec = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: globals.MetricNamespace,
+			Subsystem: clusterSubSystem,
+			Name:      "grpc_request_size_bytes",
+			Help:      "Histogram of request sizes for HTTP requests.",
+			// 100 bytes, 1kb, 10kb, 100kb, 1mb, 10mb, 100mb, 1gb
+			Buckets: gRpcMsgSizeBuckets,
+		},
+		[]string{labelGRpcCode, labelGRpcService, labelGRpcMethod},
+	)
+
+	// gRpcResponseSize collections measurements of how large each response
+	// from the boundary controller cluster is.
+	gRpcResponseSize prometheus.ObserverVec = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: globals.MetricNamespace,
+			Subsystem: clusterSubSystem,
+			Name:      "grpc_response_size_bytes",
+			Help:      "Histogram of response sizes for HTTP responses.",
+			// 100 bytes, 1kb, 10kb, 100kb, 1mb, 10mb, 100mb, 1gb
+			Buckets: gRpcMsgSizeBuckets,
+		},
+		[]string{labelGRpcCode, labelGRpcService, labelGRpcMethod},
+	)
+)
+
+func statusFromError(err error) *status.Status {
+	if s, ok := status.FromError(err); ok {
+		return s
+	}
+	return status.New(codes.Unknown, "Unknown Code")
+}
```

Comment:
This is almost identical to https://pkg.go.dev/google.golang.org/grpc/status#Convert (the difference is in the message for the unknown status), do we need our own function for this?